src/server/scene/surface_stack.cpp

```diff
@@ -493,6 +493,24 @@ void ms::SurfaceStack::send_to_back(const mir::scene::SurfaceSet &ss)
     }
 }
 
+auto ms::SurfaceStack::z_order(std::weak_ptr<scene::Surface> const& surface) const -> unsigned int
+{
+    RecursiveReadLock lg(guard);
+    if (surface.expired())
+        return 0;
+
+    auto const lock_surface = surface.lock();
+    auto const& layer = surface_layers[lock_surface->depth_layer()];
+    for (size_t surface_index = 0; surface_index < layer.size(); surface_index++)
+    {
+        auto const& other = layer[surface_index];
+        if (surface.lock() == other)
+            return surface_index;
+    }
+
+    return {};
```

Comment:
Nit: This was a little unexpected, especially since 0 is returned explicitly above.

Reply:
Whoops! Leftover, good catch!